utils/lib/CommonBuild.hs

```diff
@@ -150,16 +154,46 @@ weederStep dryRun = do
 ----------------------------------------------------------------------------
 -- Stack Haskell Program Coverage and upload to Coveralls
 
+-- | Name of the environment variable containing the coveralls.io token, which
+-- is used for uploading coverage information to coveralls.io.
+--
+-- If coverage information shouldn't be uploaded, or such token is not
+-- available, then the string can be set to "".
+newtype CoverallsTokenEnvVar = CoverallsTokenEnvVar Text deriving (Show)
+
+-- | Extra arguments to be passed to the @shc@ program, which provides
+-- coveralls.io integration to @stack@ based projects.
+newtype ExtraShcArgs = ExtraShcArgs [Text] deriving (Show)
+
+-- | Path to the directory where the coverage files (.tix) can be found.
+newtype TixDirectory = TixDirectory FilePath deriving (Show)
```

Comment:
Note that this isn't what `tixDir` is used for down there :upside_down_face: 
The tix files can be found in `stack path --local-hpc-root` which is basically what `shc` uses under the hood. 

Here however, we are looking for extra tix files possibly generated by executables built with `--coverage` and ran during the test suite of another package! For example, in the case of `cardano-wallet`, during the integration tests, we actually make calls to the `cardano-wallet` command-line to have automated tests of this command line. This is done via the [process](https://hackage.haskell.org/package/process-1.6.6.0) library and therefore, execution of commands happen in _another process_ that the one executing the test suite. Therefore, `shc` and `stack` and not able to capture this properly. Yet, if the command-line is built using `--coverage` (and this is why this option is passed during the `build` step), it'll generate a corresponding `.tix` file at the root of the project. 

This extra `.tix` tile can then be passed to `shc` when using the `combined` and `custom` options (combined means, all "normally" generated .tix files for the project located in `stack path --local-hpc-root` plus, the extra tix files given as arguments. 

I believe it is possible to figure out what are the roots of each package in the project (in our case, it was simply every folder in `lib/`) from stack or a `cabal.project` or simply from the `stack.yaml` and remove the need for this option altogether :thinking: ?